### DISCUSSION TITLE
Push doesn't trigger workflow action in an open PR

### DISCUSSION THREAD

**Discussion by amete** [community/community#25332] [Actions]
<p>Hi guys,</p>
<p>I’m trying to get accustomed to GitHub actions. Probably I’m missing something obvious. I have two workflows, one called <code>CI</code> and the other called <code>Clang-Format</code>, configured to trigger as:</p>
<pre><code class="lang-auto">name: CI
on:
  push:
    branches: [ master ]
  pull_request:
    branches: [ master ]
</code></pre>
<pre><code class="lang-auto">name: Clang-Format

on: push
</code></pre>
<p>Now, in an open PR (to master), when I push any changes to the relevant branch, both <code>CI</code> and <code>Clang-Format</code> triggers. Former due to <code>pull_request</code> and the latter due to <code>push</code>. So far so good.</p>
<p>Then, the latter commits changes and pushes to the same branch (within the action). However, this latter push doesn’t trigger the <code>CI (pull_request)</code> but it triggers Travis which is also configured for pull request.</p>
<p>I have a very basic example here: <a href="https://github.com/amete/hello_world/pull/6" rel="nofollow noopener">https://github.com/amete/hello_world/pull/6</a></p>
<p>Why doesn’t the last commit/push trigger the <code>CI/build (pull_request)</code>? Any ideas?</p>
<p>Many thanks.</p>
<p>Best,<br>
Serhan</p>

**Comment by weide-zhou** [comment, marked as answer]
<p>Hi <a class="mention" href="/amete">@amete</a>,</p>
<p>Glad to see you in Github Community Forum!</p>
<p>This is by designed. The latter commit is pushed within an action. There is a limitation of workflow:  <em>An action in a workflow run can’t trigger a new workflow run</em></p>
<p>When you use GITHUB_TOKEN in your actions, all of the interactions with the repository are on behalf of the Github-actions bot. The operations act by Github-actions bot cannot trigger a new workflow run.<br>
<img src="https://user-images.githubusercontent.com/52772761/181090469-14ab0382-ce97-4a3b-88c2-7d25c5420e7d.png" alt="lattercommit" data-base62-sha1="hDpuo5fZaokRoD2u2SG3eEiRdy7" width="686" height="197"></p>
<p>As an alternative, you can create a personal access token, and use it to create an encrypted secret, use the secret instead of GITHUB_TOKEN for the action. Please check similar <a href="https://github.community/t/github-actions-workflow-not-triggering-with-tag-push/17053/2">ticket</a> for the details.</p>
<p>Thanks</p>

**Comment by amete** [comment]
<aside class="quote group-github-partners" data-username="weide-zhou" data-post="2" data-topic="118675">
<div class="title">
<div class="quote-controls"></div>
 weide-zhou:</div>
<blockquote>
<p>ticket</p>
</blockquote>
</aside>
<p>Thanks <a class="mention" href="/weide-zhou">@weide-zhou</a>, that’s really useful!</p>
